### PR TITLE
Fix theme inheritance for default themes

### LINKED ISSUE #5156: inherit does not work with base16_theme
### Summary

I'm following the [docs guide](https://docs.helix-editor.com/themes.html?highlight=ruler#inheritance) and created a new theme file `mytheme.toml` and placed it at `~/.config/helix/themes/mytheme.toml`:

```toml
# ~/.config/helix/themes/mytheme.toml
inherits = "base16_theme"

ui.virtual.ruler = { fg = "grey", bg = "grey" }
```

Running `:theme mytheme` in `hx myfile.json` results in this output:

```
Theme does not exist
```

I've tried running `cargo xtask themelint ~/.config/helix/themes/mytheme` but the output doesn't make sense since I expect the default theme to be correct:

```
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.selection`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.selection.primary`
/home/user/.config/helix/themes/mytheme: `ui.selection` and `ui.selection.primary` cannot be equal
/home/user/.config/helix/themes/mytheme: missing `fg` for `ui.text`
/home/user/.config/helix/themes/mytheme: missing `bg` for `ui.background`
/home/user/.config/helix/themes/mytheme: missing `fg` and `bg` for `ui.statusline`
/home/user/.config/helix/themes/mytheme: missing `fg` and `bg` for `ui.statusline.inactive`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.statusline.normal`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.statusline.insert`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.statusline.select`
/home/user/.config/helix/themes/mytheme: `ui.statusline.normal` and `ui.statusline.insert` cannot be equal
/home/user/.config/helix/themes/mytheme: `ui.statusline.normal` and `ui.statusline.select` cannot be equal
/home/user/.config/helix/themes/mytheme: missing `bg` for `ui.cursorline.primary`
/home/user/.config/helix/themes/mytheme: missing `fg` for `ui.virtual.whitespace`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.virtual.indent-guide`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.virtual.ruler`
/home/user/.config/helix/themes/mytheme: missing `fg` and `bg` for `ui.menu`
/home/user/.config/helix/themes/mytheme: missing `fg` and `bg` for `ui.help`
/home/user/.config/helix/themes/mytheme: missing `bg` for `ui.popup`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.window`
/home/user/.config/helix/themes/mytheme: missing `bg` for `ui.cursor.primary`
/home/user/.config/helix/themes/mytheme: missing `fg` or `bg` for `ui.cursor.match`
Error: "/home/user/.config/helix/themes/mytheme has issues"
```

All I want is to change the default color of the rulers since it is set to an obnoxious red by default. I picked `base16_theme` as the `inherits` because it is the output of running `:theme` in the editor. I'm working on a clone of the repo from yesterday. Please see below for the exact git commit.

### Reproduction Steps

 

I tried this:

1. `hx`

I expected this to happen:

Instead, this happened:

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

Linux

### Terminal Emulator

zsh 5.4.1 (x86_64-unknown-linux-gnu) running in tmux 3.3a

### Helix Version

22.12-25-g0f2ae35a

### PR DESCRIPTION
**Pull request by hunterliao29**
close #5156

### CODE REVIEW

File: helix-view/src/theme.rs [hunk 1]
```diff
@@ -78,11 +81,16 @@ impl Loader {
...
             })?;
 
-            let parent_theme_toml = self.load_theme(
-                parent_theme_name,
-                base_them_name,
-                base_them_name == parent_theme_name,
-            )?;
+            let parent_theme_toml = match parent_theme_name {
+                // load default themes's toml from const.
+                "Default" => toml::from_slice(DEFAULT_THEME_DATA)?,
```
Comment (Omnikar) [inline helix-view/src/theme.rs:86]:
```suggestion
                "default" => toml::from_slice(DEFAULT_THEME_DATA)?,
```

File: helix-view/src/theme.rs [hunk 2]
```diff
@@ -14,15 +14,18 @@ use toml::{map::Map, Value};
...
+const BASE16_DEFAULT_THEME_DATA: &[u8] = include_bytes!("../../base16_theme.toml");
+
 pub static DEFAULT_THEME: Lazy<Theme> = Lazy::new(|| Theme {
     name: "default".into(),
-    ..toml::from_slice(include_bytes!("../../theme.toml")).expect("Failed to parse default theme")
+    ..toml::from_slice(DEFAULT_THEME_DATA).expect("Failed to parse default theme")
 });
 
 pub static BASE16_DEFAULT_THEME: Lazy<Theme> = Lazy::new(|| Theme {
     name: "base16_theme".into(),
```
Comment (the-mikedavis) [inline helix-view/src/theme.rs:27]: A bit unrelated to this PR but this should be `"base16_default"`. There's a mismatch where you can choose this theme with `:theme base16_default` but then `:theme` (which prints the current theme) says "base16_theme".

File: helix-view/src/theme.rs [hunk 3]
```diff
@@ -14,15 +14,18 @@ use toml::{map::Map, Value};
 use crate::graphics::UnderlineStyle;
 pub use crate::graphics::{Color, Modifier, Style};
 
+// store default theme in to byte array.
+const DEFAULT_THEME_DATA: &[u8] = include_bytes!("../../theme.toml");
+const BASE16_DEFAULT_THEME_DATA: &[u8] = include_bytes!("../../base16_theme.toml");
```
Comment (the-mikedavis) [inline helix-view/src/theme.rs:19]: Is it possible to setup constants with these parsed as `toml::Value`s? That way we don't have to re-parse them when doing inherits

Comment (hunterliao29) [reply helix-view/src/theme.rs]: I followed the existing code base to use ```Lazy<Value>``` to parsed it.

File: helix-view/src/theme.rs [hunk 4]
```diff
@@ -14,15 +14,24 @@ use toml::{map::Map, Value};
 use crate::graphics::UnderlineStyle;
 pub use crate::graphics::{Color, Modifier, Style};
 
+// store default theme in to byte array.
+// const BASE16_DEFAULT_THEME_DATA: &[u8] = include_bytes!("../../base16_theme.toml");
```
Comment (the-mikedavis) [inline helix-view/src/theme.rs:18]: Looks like these comments can be removed now

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**